tests/plugins/test_store_metadata.py

```diff
@@ -140,32 +143,49 @@ def test_metadata_plugin(workflow, source_dir,
             FROM scratch
             RUN yum install -y python
             """)
+        all_parents = ['fedora', 'scratch']
     else:
         df_content = dedent("""\
             FROM fedora
             RUN yum install -y python-django
             CMD blabla
             """)
+        all_parents = ['fedora']
 
-    prepare(workflow)
-    mock_dockerfile(workflow, df_content)
-
-    dockerfile = workflow.build_dir.any_platform.dockerfile_with_parent_env(
-        workflow.imageutil.base_image_inspect()
-    )
+    prepare(workflow, init_dirs=init_dirs)
+    if init_dirs:
+        mock_dockerfile(workflow, df_content)
 
-    df_images = DockerfileImages(dockerfile.parent_images)
-    for parent in dockerfile.parent_images:
-        if parent != 'scratch':
-            df_images[parent] = "sha256:spamneggs"
+        dockerfile = workflow.build_dir.any_platform.dockerfile_with_parent_env(
+            workflow.imageutil.base_image_inspect()
+        )
 
-    env = (MockEnv(workflow)
-           .for_plugin(StoreMetadataPlugin.key)
-           .set_plugin_args({"url": "http://example.com/"})
-           .set_dockerfile_images(df_images)
-           .set_plugin_result(RPMqaPlugin.key, "rpm1\nrpm2")
-           .set_plugin_result(VerifyMediaTypesPlugin.key, verify_media_results)
-           .set_plugin_result(AddHelpPlugin.key, help_results))
+        df_images = DockerfileImages(dockerfile.parent_images)
+        for parent in dockerfile.parent_images:
+            if parent != 'scratch':
+                df_images[parent] = "sha256:spamneggs"
+    else:
+        df_images = DockerfileImages(all_parents)
+        df_images['fedora'] = "sha256:spamneggs"
+
+    if init_dirs:
+        env = (MockEnv(workflow)
+               .for_plugin(StoreMetadataPlugin.key)
+               .set_plugin_args({"url": "http://example.com/"})
+               .set_dockerfile_images(df_images)
+               .set_plugin_result(RPMqaPlugin.key, "rpm1\nrpm2")
+               .set_plugin_result(VerifyMediaTypesPlugin.key, verify_media_results)
+               .set_plugin_result(AddHelpPlugin.key, help_results)
+               .mock_build_outcome(failed=failed))
+    else:
+        env = (MockEnv(workflow)
+               .for_plugin(StoreMetadataPlugin.key)
+               .set_plugin_args({"url": "http://example.com/"})
+               .set_dockerfile_images(df_images)
+               .set_plugin_result(RPMqaPlugin.key, "rpm1\nrpm2")
+               .set_plugin_result(VerifyMediaTypesPlugin.key, verify_media_results)
+               .set_plugin_result(AddHelpPlugin.key, help_results)
+               .mock_build_outcome(failed=failed))
```

Comment:
It looks like the `if` and `else` are exactly the same code

Reply:
fixed